### ISSUE TITLE
Should check if nix-prefetch-git is at least the required version

### ISSUE THREAD

**Issue by expipiplus1** [expipiplus1/update-nix-fetchgit#10]

**Comment by DavidEGrayson** [comment]
Nix is good at doing that!  I was thinking we should aim to write a Nix expression that builds the code (or somehow uses a Hackage package for it), bundles our program with the latest version of nix-prefetch-git, and runs the tests.  Perhaps we need to make a wrapper script that has the full path of nix-prefetch-git and prepends it to the PATH: this way Nix itself knows that our program depends on nix-prefetch-git and would be guaranteed to install it, and also there is no way our program would use some bad version at runtime.  I think wrapper scripts like that are used for other projects too.

**Comment by expipiplus1** [comment]
Just putting the full path to nix-prefetch-git in the source will do this. Although this would obviously have to be done at build time as the store path isn't distributable.

The wrapper script is probably neatest though seeing as this is an executable.

**Comment by DavidEGrayson** [comment]
Yeah, either way would work.  I don't know much about Cabal/GHC but maybe we could pass a configuration option to it that says what the path of nix-prefetch-git is, and that could get baked into the executable.  In C/C++ we would just use the preprocessor to do this.
